sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java

```diff
@@ -910,15 +973,24 @@ public void startBundle() {
       public void processElement(ProcessContext ctx) {
         // Need to copy the document because mongoCollection.insertMany() will mutate it
         // before inserting (will assign an id).
+
         batch.add(new Document(ctx.element()));
         if (batch.size() >= spec.batchSize()) {
-          flush();
+          if (spec.isUpdate()) {
```

Comment:
To avoid code copying, please, adjust `flush()` method accordingly.

Reply:
@aromanenko-dev @pabloem Have checked in the changes. We have an UpdateConfiguration object where all the update related details are wrapped. Sample below:
MongoDbIO.write().withUri(options.getMongoDBUri())
                        .withDatabase(options.getDatabase())
                        .withCollection(options.getCollection()).withUpdateConfiguration(
                                UpdateConfiguration.create().withUpdateKey("accId").withUpdateFields(
                                        UpdateField.of("$set", "category", "category"),
                                        UpdateField.of("$set","balance", "balance"),
                                        UpdateField.of("$push", "transactions"))
 Last update which is a push does update the dest field with full input document. Please review and let me know.

Reply:
@aromanenko-dev @pabloem gentle reminder to review

Reply:
@pareshsarafmdb Thanks, let me take a look. 
@pabloem Do you have any comments on this?

Reply:
(Pablo is on vacation until August, please ping again in early August.)